components/server/src/server.rs

```diff
@@ -368,10 +368,18 @@ impl<ER: RaftEngine> TiKVServer<ER> {
         self.lock_files.push(f);
 
         if tikv_util::panic_mark_file_exists(&self.config.storage.data_dir) {
-            fatal!(
-                "panic_mark_file {} exists, there must be something wrong with the db.",
-                tikv_util::panic_mark_file_path(&self.config.storage.data_dir).display()
-            );
+            if self.config.allow_restart_on_data_corruption {
```

Comment:
This doesn't seem to be a good idea. Once there is such a configuration, it will be abused and we will lose the chance to find what's actually happening.

Reply:
how about make it hidden and not mention in config template and docs? For a user they see intermittent corruption error where the physical file is not actually corrupted on retry. We need a temporary workaround until we get a better fix (i.e. auto-recovery on SST corruption).

Reply:
the temporary workaround is to delete the mark file manually.  I don't think this is really necessary.

Reply:
discuss with @Connor1996 offline. It may not be a good idea to provide such an option. I will remove it.